src/plugins/customdata.ts

```diff
@@ -0,0 +1,28 @@
+// Allows data sent via. normal DOM event handlers to be logged alongside Clarity data by listening to the
+// custom DOM event "claritydata".
+import { addEvent, bind } from "./../core";
+
+// Events coming from normal DOM will have additional data in details property
+interface ICustomEvent extends Event {
+    detail: any;
```

Comment:
Created a new type to specify that Clarity requires a "type" and "data" for each custom event log. Requiring a "type" should make data easier to query for analysts.